modules/effects/src/effect_creator.ts

```diff
@@ -23,8 +23,9 @@ type ConditionallyDisallowActionCreator<DT, Result> = DT extends false
 export function createEffect<
   C extends EffectConfig & { functional?: false },
   DT extends DispatchType<C>,
-  OT extends ObservableType<DT, OT>,
-  R extends EffectResult<OT>
+  OTP,
+  R extends EffectResult<OT>,
+  OT extends ObservableType<DT, OTP>
```

Comment:
Resolves #4279
Introduce a new type to resolve work around the circular generic type `OT`

Reply:
[nit] You likely meant #4275 😅